Add makefile, pin python version

hardyoyo opened the pull request:
The .python_version file is copied from stg, the Makefile is new, will give it a spin tomorrow. Makefile works great. Ready to merge, and, indeed, has already been merged.

hardyoyo committed:
Add Makefile to faciliate easier migration runs. Pin Python version

hardyoyo committed:
resolve errors found by Checkmake in Makefile

everreau commented:
looks fine to me. I saw a comment somewhere that said it was the python version from prd but I'm guessing that was a mistaken comment (that you've corrected here) that it's from stage.  We only run journal transporter from stg.  I've run it locally in the past but there are issues that make it not worth doing for real migrations.

hardyoyo committed:
Improvements

- strip extra spaces from params before we use them
- start migration targets in the background so they don't block us
- revise curl command for prd ready check
- refactor functions for clarity

This Makefile version is tested and confirmed to work correctly for both
dev and prd migrations.

hardyoyo committed:
Fix readycheck curl for prd to exit if prd isn't reachable.

hardyoyo committed:
Fix wonky whitespace

hardyoyo committed:
Disable Makefile verbosity, add hint about dry runs
